### DISCUSSION TITLE
Many node extends ElementNode class, but no inherit ElementNode class methods, How did lexical did that?

### DISCUSSION THREAD

**Discussion by SD-Gaming** [facebook/lexical#6502] [Q&A] [answered]
I looked over the source code and found that many Node extend the ElementNode, but do not have all the ElementNode methods.

for example: 
the ElementNode implement getStyle method,
[LexicalElementNode.getStyle()](https://github.com/facebook/lexical/blob/031891eda7f37563666f3fcf0d8f18d1fa8904a7/packages/lexical/src/nodes/LexicalElementNode.ts#L104)

and  class ParagraphNode extends ElementNode,
[class ParagraphNode extends ElementNode](https://github.com/facebook/lexical/blob/9b45ce9b6709dbcb476f814de146af7ef0e79ef1/packages/lexical/src/nodes/LexicalParagraphNode.ts#L46)

but ParagraphNode not have getStyle method.

I read all that I think is relative code, but nothing founded.

could someone tell me how did lexical did that work?

**Comment by etrepum** [comment]
ElementNode is a class, not an abstract interface. These subclasses are inheriting the implementation from the base class.

**Reply by SD-Gaming** [reply]
So my question is: Why does class **ParagraphNode** extends **ElementNode** but not inherit **ElementNode** methods like getStyle(), setStyle(), etc.?

If ElementNode is an abstract class, I wouldn't ask this question.

**Reply by etrepum** [reply, marked as answer]
It does inherit those methods. Maybe you are confused because the API documentation is generated with settings that don't repeat inherited methods that are not overridden?

**Reply by SD-Gaming** [reply]
I found the problem. In v0.17.0, everything is OK.My problem is with v0.16.1.

Thanks for your inspiration!
